### ISSUE TITLE
I can't start working with jellyfin

### ISSUE THREAD

**Issue by Druidblack** [FoxxMD/multi-scrobbler#200]
**Describe the bug**
After upgrading to version 0.8.5 and using the API to work with jelly fun, I can't get playback data.

- multi-scrobbler: [e.g. 0.8.5 on docker]
- jellyfin [e.g. 10.9.11]

> [2024-10-05 19:43:00.294 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Poll retries (5) equal to max poll retries (5), stopping polling!
[2024-10-05 19:43:00.293 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot read properties of undefined (reading 'map')
TypeError: Cannot read properties of undefined (reading 'map')
    at Function.formatPlayObj (CWD/src/backend/sources/JellyfinApiSource.ts:244:44)
    at JellyfinApiSource.sessionToPlayerState (CWD/src/backend/sources/JellyfinApiSource.ts:316:51)
    at <anonymous> (CWD/src/backend/sources/JellyfinApiSource.ts:281:101)
    at Array.map (<anonymous>)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:281:87)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-05 19:43:00.292 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-05 19:42:57.467 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot read properties of undefined (reading 'map')
TypeError: Cannot read properties of undefined (reading 'map')
    at Function.formatPlayObj (CWD/src/backend/sources/JellyfinApiSource.ts:244:44)
    at JellyfinApiSource.sessionToPlayerState (CWD/src/backend/sources/JellyfinApiSource.ts:316:51)
    at <anonymous> (CWD/src/backend/sources/JellyfinApiSource.ts:281:101)
    at Array.map (<anonymous>)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:281:87)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)

![3453535345345](https://github.com/user-attachments/assets/d1011f0a-b33e-449a-928c-1c6de6a99d3f)
![34534534](https://github.com/user-attachments/assets/3e9eafa3-b18f-47f0-b061-6ecd96368638)

**Comment by FoxxMD** [comment]
Hi @Druidblack , please try using this docker image `foxxmd/multi-scrobbler:pr-205` and let me know if this fixes your issue. Thanks

**Comment by Druidblack** [comment]
![65464564565](https://github.com/user-attachments/assets/b02d9720-7223-42f5-a523-14a5ad7e2381)
`[2024-10-10 21:47:04.426 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Poll retries (5) equal to max poll retries (5), stopping polling!
[2024-10-10 21:47:04.424 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:47:04.423 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:47:04.422 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:58.663 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:58.662 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:58.661 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:50.097 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:50.097 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:50.094 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:49.412 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:49.412 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:49.411 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:41.400 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:41.400 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:41.399 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:38.390 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:38.389 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:38.388 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:21.212 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Poll retries (5) equal to max poll retries (5), stopping polling!
[2024-10-10 21:46:21.211 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:21.211 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:21.210 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:20.709 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:20.709 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:20.708 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:19.084 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:19.083 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:19.082 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:18.557 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:18.557 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:18.556 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:18.396 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:18.395 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:18.394 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:17.199 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:17.198 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:17.197 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:16.695 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:16.694 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:16.693 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:16.072 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:16.072 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:16.071 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:15.383 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:15.382 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:15.381 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy
[2024-10-10 21:46:14.545 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Cannot destructure property 'data' of '{}' as it is undefined.
TypeError: Cannot destructure property 'data' of '{}' as it is undefined.
    at buildTrackString (CWD/src/core/StringUtils.ts:62:15)
    at JellyfinApiSource.getRecentlyPlayed (CWD/src/backend/sources/JellyfinApiSource.ts:289:76)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at JellyfinApiSource.doPolling (CWD/src/backend/sources/AbstractSource.ts:389:34)
    at JellyfinApiSource.startPolling (CWD/src/backend/sources/AbstractSource.ts:324:27)
    at JellyfinApiSource.poll (CWD/src/backend/sources/AbstractSource.ts:292:9)
[2024-10-10 21:46:14.544 +0300] ERROR  : [App] [Sources] [Jellyfin - unnamed] Error occurred while polling
[2024-10-10 21:46:14.543 +0300] WARN   : [App] [Sources] [Jellyfin - unnamed] Player State for  ->  - Жизнь среди заразы <-- is being dropped because 'usersAllow does not include user iliy`

**Comment by Druidblack** [comment]
For information. With version 0.8.5, sometimes when restarting the container, Multi Scrobbler started working with jellybean, it displayed several players (with question marks) and when starting music, it showed the filler and sent data to last.fm . Restarting the container again did not work with jelly bean.

**Comment by FoxxMD** [comment]
@Druidblack can you try pulling the same docker image again, see if the new changes make a difference.

**Comment by jpjpjunior** [comment]
I can confirm that with this new docker image, jellyfin scrobbling is working again. I was also affected by the recent update and was waiting for a fix. I have only been test this interim update for about 10 minutes, but can confirm it is polling my jellyfin instance correctly.

**Comment by jpjpjunior** [comment]
> @Druidblack can you try pulling the same docker image again, see if the new changes make a difference.

However, After closer inspection, my json config files ( listenbrainz, and a subsonic client) are returning an error. 
![Screenshot from 2024-10-10 16-41-26](https://github.com/user-attachments/assets/079a2612-58b3-48af-922c-9cdc05a6c320)

**Comment by Druidblack** [comment]
I'll confirm it. Jelly bean works with the new version. But in the version (
pr205-c2b294f91b647a751bd02e60a29259bff8848439) Lastfm (source) is missing.

**Comment by FoxxMD** [comment]
@jpjpjunior @Druidblack please try re-pulling `foxxmd/multi-scrobbler:pr-205` as I've updated it with additional fixes.

**Comment by jpjpjunior** [comment]
> @jpjpjunior @Druidblack please try re-pulling `foxxmd/multi-scrobbler:pr-205` as I've updated it with additional fixes.
![Screenshot from 2024-10-11 14-23-41](https://github.com/user-attachments/assets/f6e6cfdf-d55e-43a3-9d40-5cdb0461ee19)
![Screenshot from 2024-10-11 14-23-08](https://github.com/user-attachments/assets/54ec6da6-2ef1-4a05-a8b7-8ddb3c1b8214)

Issue still persists for me.

**Comment by FoxxMD** [comment]
@jpjpjunior are you sure you pulled the new changes? Can you confirm that on the dashboard or in the logs it says Version `pr205-77c9141d923157634a775586e4f69cbab8fe1ff6` ?

If the version is correct can you please post how your configs are setup? Are you using files or ENV? Samples of files?

**Comment by jpjpjunior** [comment]
> @jpjpjunior are you sure you pulled the new changes? Can you confirm that on the dashboard or in the logs it says Version `pr205-77c9141d923157634a775586e4f69cbab8fe1ff6` ? If there are still errors they should look very different than that screenshot as the library shown in the stacktrace is no longer installed in MS.
> 
> If the version is correct can you please post how your configs are setup? Are you using files or ENV? Samples of files?

I am running both ENV variables and config files. Mainly because listenbrainz cannot be used in ENV. Should i just run everything in config json files? 


![Screenshot from 2024-10-11 14-38-29](https://github.com/user-attachments/assets/dde491ce-0eca-4558-8dcb-280ccdc766a0)

**Comment by FoxxMD** [comment]
I don't think you have pulled the latest image. The version is wrong in MS. The version is controlled by a dockerfile ARG that is set during publishing so it can be traced in the [image digest](https://hub.docker.com/layers/foxxmd/multi-scrobbler/pr-205/images/sha256-30b3a6b00a0f90011463df28413512c03824b72cc4222f12a8651954daf14364?context=repo), from the latest pr-205 image:

![image](https://github.com/user-attachments/assets/e95a8552-3078-4f50-a4f2-50966212aa04)

Please try updating your image...if you are using the default (not ghcr.io) you should see the digest as `sha256:f722715414b99...`

To update the image using plain docker run `docker image pull foxxmd/multi-scrobbler:pr-205`

```
$ docker image pull foxxmd/multi-scrobbler:pr-205
pr-205: Pulling from foxxmd/multi-scrobbler
Digest: sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0
Status: Image is up to date for foxxmd/multi-scrobbler:pr-205
docker.io/foxxmd/multi-scrobbler:pr-205
```

To update with docker-compose run `docker compose pull`

To verify you have the updated image run `docker inspect foxxmd/multi-scrobbler:pr-205`, you should see
```
"RepoDigests": [
            "foxxmd/multi-scrobbler@sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0"
        ],
```

in the output

**Comment by Druidblack** [comment]
![Безымянный](https://github.com/user-attachments/assets/9294e01c-d3ef-481c-91c5-20a49b4f6339)
Works.

**Comment by jpjpjunior** [comment]
> I don't think you have pulled the latest image. The version is wrong in MS. The version is controlled by a dockerfile ARG that is set during publishing so it can be traced in the [image digest](https://hub.docker.com/layers/foxxmd/multi-scrobbler/pr-205/images/sha256-30b3a6b00a0f90011463df28413512c03824b72cc4222f12a8651954daf14364?context=repo), from the latest pr-205 image:
> 
> ![image](https://private-user-images.githubusercontent.com/4663766/375858343-e95a8552-3078-4f50-a4f2-50966212aa04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NzQ0MDgsIm5iZiI6MTcyODY3NDEwOCwicGF0aCI6Ii80NjYzNzY2LzM3NTg1ODM0My1lOTVhODU1Mi0zMDc4LTRmNTAtYTRmMi01MDk2NjIxMmFhMDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTFUMTkxNTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjgyZjY3Y2EyN2Q4ZDExYjU3YTJmODFiZWJlMDQ2OWYzYzQxNTlhMTU0Yzc0OTk3YjZlNTJkOGUwZGYxYWM0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FRihOODd3L8qtY_l94eeiOMPVo4Agen2nLCtKBUbxyI)
> 
> Please try updating your image...if you are using the default (not ghcr.io) you should see the digest as `sha256:fe3d2a27a0704ef7905acf...` or the correct version in the image data
> 
> To update the image using plain docker run `docker image pull foxxmd/multi-scrobbler:pr-205`
> 
> ```
> $ docker image pull foxxmd/multi-scrobbler:pr-205
> pr-205: Pulling from foxxmd/multi-scrobbler
> Digest: sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0
> Status: Image is up to date for foxxmd/multi-scrobbler:pr-205
> docker.io/foxxmd/multi-scrobbler:pr-205
> ```
> 
> To update with docker-compose run `docker compose pull`
> 
> To verify you have the updated image run `docker inspect foxxmd/multi-scrobbler:pr-205`, you should see
> 
> ```
> "RepoDigests": [
>             "foxxmd/multi-scrobbler@sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0"
>         ],
> ```
> 
> in the output
> 
> or run `docker inspect foxxmd/multi-scrobbler:pr-205 | grep APP_VERSION` which should output the correct version `APP_VERSION=pr205-77c9141d923157634a775586e4f69cbab8fe1ff6`
> 
> You can also try pulling the specific digest rather than the pr tag, if none of the above is working for you for some reason:
> 
> ```
> foxxmd/multi-scrobbler@sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0
> ```

My apologies .. Ill check

**Comment by Druidblack** [comment]
> > Я не думаю, что вы скачали последнюю версию образа. Версия в MS неверная. Версия контролируется аргументом dockerfile, который задается во время публикации, поэтому ее можно отследить в [дайджесте образа](https://hub.docker.com/layers/foxxmd/multi-scrobbler/pr-205/images/sha256-30b3a6b00a0f90011463df28413512c03824b72cc4222f12a8651954daf14364?context=repo) последней версии pr-205:
> > ![изображение](https://private-user-images.githubusercontent.com/4663766/375858343-e95a8552-3078-4f50-a4f2-50966212aa04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NzQ0MDgsIm5iZiI6MTcyODY3NDEwOCwicGF0aCI6Ii80NjYzNzY2LzM3NTg1ODM0My1lOTVhODU1Mi0zMDc4LTRmNTAtYTRmMi01MDk2NjIxMmFhMDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTFUMTkxNTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjgyZjY3Y2EyN2Q4ZDExYjU3YTJmODFiZWJlMDQ2OWYzYzQxNTlhMTU0Yzc0OTk3YjZlNTJkOGUwZGYxYWM0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FRihOODd3L8qtY_l94eeiOMPVo4Agen2nLCtKBUbxyI)
> > Пожалуйста, попробуйте обновить изображение... если вы используете изображение по умолчанию (а не ghcr.io), вы должны увидеть дайджест `sha256:fe3d2a27a0704ef7905acf...` или правильную версию в данных изображения
> > Чтобы обновить изображение с помощью обычного docker, запустите `docker image pull foxxmd/multi-scrobbler:pr-205`
> > ```
> > $ docker image pull foxxmd/multi-scrobbler:pr-205
> > pr-205: Pulling from foxxmd/multi-scrobbler
> > Digest: sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0
> > Status: Image is up to date for foxxmd/multi-scrobbler:pr-205
> > docker.io/foxxmd/multi-scrobbler:pr-205
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Для обновления с помощью docker-compose run `docker compose pull`
> > Чтобы убедиться, что у вас запущено обновленное изображение `docker inspect foxxmd/multi-scrobbler:pr-205`, вы должны увидеть
> > ```
> > "RepoDigests": [
> >             "foxxmd/multi-scrobbler@sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0"
> >         ],
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > в выходных данных
> > или запустить `docker inspect foxxmd/multi-scrobbler:pr-205 | grep APP_VERSION` который должен вывести правильную версию `APP_VERSION=pr205-77c9141d923157634a775586e4f69cbab8fe1ff6`
> > Вы также можете попробовать использовать конкретный дайджест, а не тег pr, если по какой-либо причине у вас ничего из вышеперечисленного не работает:
> > ```
> > foxxmd/multi-scrobbler@sha256:fe3d2a27a0704ef7905acff512d88dc64d3e6c93d3006450996e8e32ab17dcf0
> > ```
> 
> Приношу свои извинения .. Я проверю

Delete the image that you are using in docker. Since the version of container 205 remains unchanged, when starting docker, the new image is not downloaded, the old one is used. It needs to be deleted so that the docker downloads container 205 with the latest fixes again.

**Comment by jpjpjunior** [comment]
works

**Comment by FoxxMD** [comment]
These fixes are in [0.8.6](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.6) (`latest` docker image)
